src/Game.ts

```diff
@@ -11,15 +13,34 @@ class Game {
         this.setup();
         this.turn = 'white';
     }
-
+    
     afterMove(field: Field, move: string) {
-        this.movePiece(field, move);
+        const newField = this.board.getField(parseInt(move[0]), parseInt(move[2]));
+        this.movePiece(field, newField);
         this.changeTurn();
+        this.promotePawn(newField);
         // Logika która powinna znajdować sie po ruchu znajduje się tutaj,
         // oczywiście chodzi tutaj o wywołania odpowiednich funkcji tylko :)
         // czyli np. sprawdzenie czy jest szach, mat, pat, zmiana tury itp.
     }
 
+    promotePawn(newField: Field): void {
```

Comment:
Może skróćmy tą metodę
zrobmy warunek: 
const color = this.turn==='white' ? 0 : 7
if((this.board.fields[color][y].piece instanceof Pawn) {
                    this.board.fields[color][y].piece = new Queen(this.turn);
                    this.board.fields[color][y].piece?.render(newField);
}

Dzięki temu pozbywamy się całego drugiego fora

Reply:
dzięki! 
zrobiłam tak jak piszesz, ale dodałam jeszcze jedna zmienną, bo this.turn po ruchu białych zwraca "black", a po czarnych "white", więc po dojściu białego piona do końca pion zmieniał się w czarną królową, zamiast w białą, po dodaniu zmiennej działa poprawnie, nie miałam na to lepszego pomysłu

Reply:
Jeżeli zamienimy miejscami te dwie funkcje:
        this.changeTurn();
        this.promotePawn(newField);
 Wszystko bedzie dzialac prawidlowo i nie bedzie potrzebna zmiennna a tez z logicznego punktu widzenia najpierw zamieniami na figure a dopiero pozniej zmieniamy ture

Reply:
no racja, dzięki :)